lib/src/main/java/com/diffplug/spotless/java/IdeaFormatterFunc.java

```diff
@@ -0,0 +1,93 @@
+package com.diffplug.spotless.java;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.diffplug.spotless.ForeignExe;
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.ProcessRunner;
+
+public final class IdeaFormatterFunc implements FormatterFunc.NeedsFile {
+
+	private static final Logger LOGGER =
+			LoggerFactory.getLogger(IdeaStep.class);
+
+	private static final String DEFAULT_IDEA = "idea";
+
+	private String binaryPath;
+	private String configPath;
+	private boolean withDefaults;
+
+	private IdeaFormatterFunc(boolean withDefaults, String binaryPath,
+			String configPath) {
+		this.withDefaults = withDefaults;
+		this.configPath = configPath;
+		this.binaryPath = Objects.requireNonNullElse(binaryPath, DEFAULT_IDEA);
+		resolveFullBinaryPathAndCheckVersion();
+	}
+
+	private void resolveFullBinaryPathAndCheckVersion() {
+		var exe = ForeignExe.nameAndVersion(this.binaryPath, "IntelliJ IDEA")
+				.versionRegex(Pattern.compile("(IntelliJ IDEA) .*"))
+				.fixCantFind("IDEA executable cannot be found on your machine, "
+						+ "please install it and put idea binary to PATH; or report the problem")
+				.fixWrongVersion("Provided binary is not IDEA, "
+						+ "please check it and fix the problem; or report the problem");
+		try {
+			this.binaryPath = exe.confirmVersionAndGetAbsolutePath();
+		} catch (IOException e) {
+			throw new IllegalArgumentException("binary cannot be found", e);
+		} catch (InterruptedException e) {
+			throw new IllegalArgumentException(
+					"binary cannot be found, process was interrupted", e);
+		}
```

Comment:
This should happen when `IdeaStep` is being created, see here for an example

https://github.com/diffplug/spotless/blob/14204e4d39b6b389c8cfee0b2a70cd94e5579be6/lib/src/main/java/com/diffplug/spotless/go/GofmtFormatStep.java#L69-L80

Reply:
Greatly appreciate, thanks!

I tried to make IDEA lazy.